README.md

```diff
@@ -8,92 +8,231 @@
 
-df = DataFrame(x = 1:3, y = [2, 1, 2])
-x = [2, 1, 0]
+    This rule applies to all DataFramesMeta macros.
 
-@with(df, :y .+ 1)
-@with(df, :x + x)  # the two x's are different
+## `@select`
 
-x = @with df begin
-    res = 0.0
-    for i in 1:length(:x)
-        res += :x[i] * :y[i]
-    end
-    res
-end
+Column selections and transformations. Only newly created columns are kept. 
+Operates on both a `DataFrame` and a `GroupedDataFrame`. 
 
-@with(df, df[:x .> 1, ^(:y)]) # The ^ means leave the :y alone
+When given a `GroupedDataFrame`, performs a transformation by group and then 
+"spreads" the result to a vector of the same length as vectors in the input 
+data frame. 
 
+```julia
+df = DataFrame(x = [1, 1, 2, 2], y = [1, 2, 101, 102]);
+gd = groupby(df, :x);
+@select(df, :x, :y)
+@select(df, x2 = 2 * :x, :y)
+@select(gd, x2 = 2 .* :y .* first(:y))
 ```
+## `@transform`
 
-`@with` is the fundamental macro used by the other metaprogramming
-utilities.
+Add additional columns based on keyword arguments. Operates on both a 
+`DataFrame` and a `GroupedDataFrame`. 
 
-`@with` creates a function, so scope within `@with` is a local scope.
-Variables in the parent can be read. Writing to variables in the parent scope
-differs depending on the type of scope of the parent. If the parent scope is a
-global scope, then a variable cannot be assigned without using the `global` keyword.
-If the parent scope is a local scope (inside a function or let block for example),
-the `global` keyword is not needed to assign to that parent scope.
+When given a `GroupedDataFrame`, performs a transformation by group and then 
+"spreads" the result to a vector of the same length as vectors in the input 
+data frame. 
+
+```julia
+df = DataFrame(x = [1, 1, 2, 2], y = [1, 2, 101, 102]);
+gd = groupby(df, :x);
+@transform(df, :x, :y)
+@transform(df, x2 = 2 * :x, :y)
+@transform(gd, x2 = 2 .* :y .* first(:y))
+```
 
 ## `@where`
 
-Select row subsets.
+Select row subsets. Operates on both a `DataFrame` and a `GroupedDataFrame`. 
 
 ```julia
+df = DataFrame(x = [1, 1, 2, 2], y = [1, 2, 101, 102]);
+gd = groupby(df, :x);
+outside_var = 1;
 @where(df, :x .> 1)
-@where(df, :x .> x)
-@where(df, :x .> x, :y .== 3)  # the two expressions are "and-ed"
+@where(df, :x .> outside_var)
+@where(df, :x .> outside_var, :y .< 102)  # the two expressions are "and-ed"
+@where(gd, :x .> mean(:x))
 ```
 
-## `@select`
+## `@combine`
+
+Summarize, or collapse, a grouped data frame by performing transformations at the group level and 
+collecting the result into a single data frame. Also works on a `DataFrame`, which 
+acts like a `GroupedDataFrame` with one group. 
 
-Column selections and transformations.
+Requires a `DataFrame` or `GroupedDataFrame` as the first argument, unlike 
+`combine` from DataFrames.jl. For example, the following code runs:
+
+```
+df = DataFrame(x = [1, 1, 2, 2], y = [1, 2, 101, 102]);
+gd = groupby(df, :x);
+combine(d -> (a = sum(d.x), b = sum(d.y)), gd)
+```
```

Comment:
I'm not sure it's really worth showing this, as it's about DataFrames and makes it harder to get what works in DataFramesMeta. I think the text below clear enough about how to swap arguments to get this to work.

Reply:
Okay i've re-organized in response.